contrib_bots/lib/hangman.py

```diff
@@ -0,0 +1,167 @@
+                       "\n           |\n           |\n============="),
+                      ("```\n   +-------+\n           |\n           |\n     " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n           |\n     " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n     " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n   | " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /| " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                       "\      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                       "\      |\n  /        |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                       "\      |\n  / \      |\n           |\n=============")]
+            outcur = []
+            wrongguess = []
+            for i in range(0, len(bot_self.currentword)):
+                if bot_self.currentword[i] in bot_self.guessed:
+                    outcur.append(str(bot_self.currentword[i]))
+                else:
+                    outcur.append("_")
+            if "_" not in outcur:
+                end = "\n\n:party_popper: Congratulations! You won! " \
+                    "To play again, type @hangman new. :party_popper:"
+                bot_self.ended = True
+            for i in range(0, len(bot_self.guessed)):
+                if bot_self.guessed[i] not in bot_self.currentword:
+                    wrongguess.append(str(bot_self.guessed[i]))
+            bot_self.output = (stages[bot_self.stage] + "\n```\n\n```" +
+                               "  ".join(outcur) + "```\n\n```Wrong guesses:" +
+                               " " + ", ".join(wrongguess) + "```" + end)
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content'].lower()
+        original_content_split = original_content.split(" ")
+        if original_content_split[1] == "new":
+            HangmanHandler.get_new_word(self)
+        elif original_content_split[1] == "guess":
+            try:
+                self.currentword
+            except AttributeError:
+                self.output = "There is currently no Hangman game " \
+                    " active! To start a new game, type `@hangman new`."
+            else:
+                HangmanHandler.process_guess(self, original_content)
```

Comment:
Same here. `process_guess` appears to be an instance method and can be called like `self.process_guess(original_content)`; `self` is passed to the instance methods implicitly.